contrib/babelfishpg_tsql/src/pl_exec.c

```diff
@@ -9515,6 +9515,13 @@ get_cast_hashentry(PLtsql_execstate *estate,
 
 		/* Note: we don't bother labeling the expression tree with collation */
 
+		/* 
+		 * Override funcresulttype with domain OID when target column datatype is 
+		 * [n][var]char or a user defined dataype created over it.
+		 */
+		if (cast_expr != NULL)
+			(*common_utility_plugin_ptr->handle_type_and_collation)(cast_expr, dsttype, InvalidOid);
```

Comment:
How are we making sure that it doesn't affect any return descriptor? For example, view with some cast around n types?